scmailclient/scmail.py

```diff
@@ -0,0 +1,493 @@
+    help="The fingerprint of the yourself mailbox.",
+)
+@click.option(
+    "--sender-fingerprint",
+    "-s",
+    prompt="Enter the fingerprint of sender",
+    callback=check_fingerprint,
+    required=False,
+    default="",
+    type=str,
+    help="The senders fingerprint.",
+)
+@click.password_option(
+    "--password",
+    "-p",
+    prompt="Enter password of private key",
+    help="The passphrase of private key",
+)
+@click.pass_context
+def retrieve(ctx, fingerprint, sender_fingerprint, password):
+    """Retrieve and Post messages from API"""
+    # Prepare send data.
+    logging.debug(fingerprint, sender_fingerprint)
+    payload = {"fingerprint": fingerprint}
+    if sender_fingerprint:
+        payload.update({"sender_fingerprint": sender_fingerprint})
+
+    # Retrieve from api.
+    r = requests.post(SECUREMAILBOX_URL + "/retrieve/", json=payload)
+    res = r.json()
+
+    if r.status_code == 200:
+        logging.info("The message retrieve successful.")
+    else:
+        logging.error(
+            f'The message retrieve fail.\nError {r.status_code} is: {res.get("error")}'
+        )
+        return
+
+    # load messages.
+    logging.debug(f"response is: {res}")
+
+    # if no message retrieve, directly return.
+    if res.get("data").get("count") == 0:
+        logging.info("Retrieve message successful. No message available.")
+        return
+
+    # all messages should be a list of message come from api.
+    all_messages = res.get("data").get("messages")
+    all_messages = [message.get("message") for message in all_messages]
+    logging.debug(all_messages)
+
+    # Decrypt the messages.
+    ok, messages = ctx.parent.gpg.decrypt_message(
+        messages=all_messages, passphrase=password
+    )
+    logging.debug(messages)
+
+    # Print all message.
+    ctx.parent.pp.pprint(messages)
+
+    if ok is False:
+        logging.error(f"Some messages decrypt fail.")
+    else:
+        logging.info(f"Decrypt {len(messages)} message successful.")
+
+
+@click.command()
+@click.option(
+    "--sender-fingerprint",
+    "-s",
+    prompt="The sender fingerprint",
+    callback=check_fingerprint,
+    required=True,
+    type=str,
+    help="The fingerprint of the sender.",
+)
+@click.option(
+    "--recipient",
+    "-r",
+    prompt="The recipient fingerprint",
+    callback=check_fingerprint,
+    required=True,
+    type=str,
+    help="The fingerprint of the recipient.",
+)
+@click.option(
+    "--message",
+    "-m",
+    required=True,
+    prompt="The message",
+    type=str,
+    help="Message that will send.",
+)
+@click.pass_context
+def send(ctx, sender_fingerprint, recipient, message):
+    """Send a message.
+
+    recipient: the fingerprint of recipient
+
+    message: the messages.
+    """
+    # Choose recipient and Encrypted
+    ok, encrypted_message = ctx.parent.gpg.encrypt_message(message, recipient)
+    if ok is False:
+        logging.error(f"Message encrypt fail.\n{encrypted_message}")
```

Comment:
Should this return if the message encryption failed?

Reply:
Yes, I think it should.